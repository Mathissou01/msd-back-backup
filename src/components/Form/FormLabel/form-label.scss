@use "src/styles/01-settings/settings.colors" as colors;

.c-FormLabel {
  display: flex;
  flex-direction: column;
  gap: 8px;
  margin-bottom: 2px;
  color: colors.$blue-2;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;

  &__Label {
    display: flex;
    flex-direction: row;
    flex-grow: 1;
    align-items: center;
    justify-content: space-between;
  }

  &__Secondary,
  &__Validation {
    color: colors.$gray-600;
    font-weight: 400;
    font-family: DINPro-Regular, serif;
  }

  &__Secondary {
    color: colors.$gray-800;
  }

  &__Validation {
    font-size: 12px;
    text-align: right;
  }
}
