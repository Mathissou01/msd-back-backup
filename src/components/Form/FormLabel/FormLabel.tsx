import React from "react";
import "./form-label.scss";

interface IFormLabelProps {
  label: string;
  secondaryLabel?: string;
  validationLabel?: string;
  forId?: string;
  tagType?: "label" | "legend";
  isRequired?: boolean;
}

export default function FormLabel({
  label,
  secondaryLabel,
  validationLabel,
  forId,
  tagType = "label",
  isRequired = false,
}: IFormLabelProps) {
  const Tag = tagType;

  return (
    <Tag className="c-FormLabel" htmlFor={forId}>
      <div className="c-FormLabel__Label">
        <span>{`${label}${isRequired ? " *" : ""}`}</span>
        {validationLabel && (
          <span className="c-FormLabel__Validation">{validationLabel}</span>
        )}
      </div>
      {secondaryLabel && (
        <div className="c-FormLabel__Label">
          <span className="c-FormLabel__Secondary">{secondaryLabel}</span>
        </div>
      )}
    </Tag>
  );
}
