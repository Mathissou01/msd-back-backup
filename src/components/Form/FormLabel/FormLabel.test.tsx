import { render, screen } from "@testing-library/react";
import React from "react";
import FormLabel from "./FormLabel";

const mock = {
  name: "form-label",
  label: "form label",
  secondaryLabel: "secondary label",
};

it("renders", () => {
  const { container } = render(
    <FormLabel
      label={mock.label}
      secondaryLabel={mock.secondaryLabel}
      forId={mock.name}
      isRequired={false}
    />,
  );

  const label = screen.getByText("form label");
  expect(label).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
