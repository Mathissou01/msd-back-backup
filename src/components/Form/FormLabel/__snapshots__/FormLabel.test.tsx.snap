// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel renders 1`] = `
<div>
  <label
    class="c-FormLabel"
    for="form-label"
  >
    <div
      class="c-FormLabel__Label"
    >
      <span>
        form label
      </span>
      <span
        class="c-FormLabel__Validation"
      >
        validation label
      </span>
    </div>
    <div
      class="c-FormLabel__Label"
    >
      <span
        class="c-FormLabel__Secondary"
      >
        secondary label
      </span>
    </div>
  </label>
</div>
`;
